### PR TITLE
Fetch/verify ERC-20 contract balances for light client wallet

### PR DESCRIPTION
**Pull request by g11tech**
**Motivation**
Light client post merge, can also fetch and verify any token balances for the contracts who's balance mapping structure is known. For well know  contracts this can be figured out.

This PR adds the capability to allow fetching,verifying and displaying ERC-20 contract balances for any ethereum address. It seeds balance mapping structure for some well know contracts.

TODOs
- [x] Investigate and fix why verify proof is failing on the token data
- [x] Deploy an ERC 20 contract on kiln and mint some token to an addresse and get the correct balance onto the lightclient display
- [x] fix the account and token balances display
- [x] figureout and seed data for some well know contracts
- [x] allow adding any token with `address` => `uint` balances mapping
**Description**
Example fetch of an ERC 20 contract balance:

image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -388,11 +478,44 @@ async function fetchAndVerifyAddress({
...
+    // Verify the proof, web3 converts nonce and balance into number strings, however
+    // ethereumjs verify proof requires them in the original hex format
+    contractProof.nonce = numberToHex(contractProof.nonce);
+    contractProof.balance = numberToHex(contractProof.balance);
+
+    verified =
+      verified &&
+      stateRoot === contractProofStateRoot &&
+      contractProof.storageProof[0]?.key === balanceSlot &&
+      (await stateManager.verifyProof(contractProof));
```
Comment (dapplion) [inline src/App.tsx:405]: This is async for local cyrpto ops or it does a remote call?

Comment (g11tech) [reply src/App.tsx:405]: async for local, it does some Trie traversal/walk, i think to load the proof, web3 is the one making any remote requests to fetch proofs

**Comment by g11tech** [comment]
Add/remove any erc20 token (or any token with any `address` =>`uint` balances mapping)

image

**Comment by g11tech** [comment]
Drop down selection for well known token (only added DAI for now)

image

**Review by dapplion — APPROVED** [review]
Let's goo! Thank you so much for this push :fire:

**Pull request merged**